src/sdltiles.cpp

```diff
@@ -924,6 +925,12 @@ void cata_tiles::draw_om( const point &dest, const tripoint_abs_omt &center_abs_
                                      0, 0, ll, false );
             }
 
+            std::vector<tripoint_abs_omt> &revealed_highlights = get_avatar().map_revealed_omts;
+            auto it = std::find( revealed_highlights.begin(), revealed_highlights.end(), omp );
+            if( blink && show_map_revealed && it != revealed_highlights.end() ) {
+                draw_from_id_string( "highlight", omp.raw(), 0, 0, lit_level::LIT, false );
+            }
+
```

Comment:
It should be something like this to avoid iterating through `map_revealed_omts` when overmap uistate does not imply showing map revealed omts:

```C++
            if( blink && show_map_revealed ) {
                if( it != revealed_highlights.end() ) {
                    std::vector<tripoint_abs_omt> &revealed_highlights = get_avatar().map_revealed_omts;
                    auto it = std::find( revealed_highlights.begin(), revealed_highlights.end(), omp );
                    draw_from_id_string( "highlight", omp.raw(), 0, 0, lit_level::LIT, false );
                }
            }

```

Reply:
The real problem is using `std::find` on a vector of points. This is simply O(n²), and will always be slow. The points need to be stored in a hash table (or a set, which uses a hash table internally) so that testing for membership can be O(1).

Reply:
There is no need to search for anything if data would not even be used.

Reply:
Oh, that is certainly true as well. It’s just that the time is dominated by searching the whole list for every single tile that might be drawn.